src/Processors/Formats/Impl/ORCBlockInputFormat.cpp

```diff
@@ -39,23 +38,53 @@ Chunk ORCBlockInputFormat::generate()
     if (!file_reader)
         prepareReader();
 
-    if (stripe_current >= stripe_total)
-        return res;
+    while (true)
+    {
+        if (!batch_reader)
+        {
+            arrow::Status reader_status = file_reader->NextStripeReader(
+                DBMS_DEFAULT_BUFFER_SIZE, include_indices, &batch_reader);
+            if (!reader_status.ok())
+                throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA,
+                                       "Failed to create batch reader: {}",
+                                       reader_status.ToString());
+        }
 
-    std::shared_ptr<arrow::RecordBatch> batch_result;
-    arrow::Status batch_status = file_reader->ReadStripe(stripe_current, include_indices, &batch_result);
-    if (!batch_status.ok())
-        throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA,
-                               "Error while reading batch of ORC data: {}", batch_status.ToString());
+        if (!batch_reader)
+            break;
```

Comment:
I think it would be better to move this under `if` in line 43 after checking `reader_status`